test/e2e/upgrade/upgrade.go

```diff
@@ -468,5 +472,64 @@ func clusterUpgrade(c configv1client.Interface, config *rest.Config, version upg
 	}
 
 	framework.Logf("Completed upgrade to %s", versionString(desired))
+
+	framework.Logf("Waiting on the worker pool to be upgraded")
+	if err := wait.PollImmediate(10*time.Second, 30*time.Minute, func() (bool, error) {
+		return isWorkerMCPUpdated(dc)
+	}); err != nil {
+		return fmt.Errorf("Worker pool did not complete upgrade: %v", err)
+	}
+	framework.Logf("Worker pool completed upgrade")
+
 	return nil
 }
+
+// TODO(runcom): drop this when MCO types are in openshift/api and we can use the typed client directly
+func isWorkerMCPUpdated(dc dynamic.Interface) (bool, error) {
+	mcps := dc.Resource(schema.GroupVersionResource{
+		Group:    "machineconfiguration.openshift.io",
+		Version:  "v1",
+		Resource: "machineconfigpools",
+	})
+	workerMcp, err := mcps.Get("worker", metav1.GetOptions{})
```

Comment:
Probably better to get all pools, and verify all of them are up to date (so loop over all).  Also, you are going to get errors due to connections dropping out so you should probably avoid returning err here (log, `return false, nil`).

Reply:
why would we need the master pool? that's checked in the MCO and reported to the CVO so the upgrade just fails

Reply:
because we can run upgrade tests on clusters that aren't just masters and workers.  We might have hundreds of worker pools (we plan on having long running clusters with at least 2-3 sets of nodes).  So you need to verify all pools are upgraded, not just a hardcoded name.

Reply:
Oh I get it now, I thought you were talking specifically about the master pool. So yeah, I can change this to list over all pools.

Reply:
thanks

Reply:
fixed now